src/integration-test/java/com/commercetools/sync/integration/externalsource/cartdiscounts/CartDiscountSyncIT.java

```diff
@@ -0,0 +1,643 @@
+import static com.commercetools.sync.integration.commons.utils.CartDiscountITUtils.CART_DISCOUNT_CART_PREDICATE_2;
+import static com.commercetools.sync.integration.commons.utils.CartDiscountITUtils.CART_DISCOUNT_DESC_1;
+import static com.commercetools.sync.integration.commons.utils.CartDiscountITUtils.CART_DISCOUNT_DESC_2;
+import static com.commercetools.sync.integration.commons.utils.CartDiscountITUtils.CART_DISCOUNT_DRAFT_2;
+import static com.commercetools.sync.integration.commons.utils.CartDiscountITUtils.CART_DISCOUNT_KEY_1;
+import static com.commercetools.sync.integration.commons.utils.CartDiscountITUtils.CART_DISCOUNT_KEY_2;
+import static com.commercetools.sync.integration.commons.utils.CartDiscountITUtils.CART_DISCOUNT_TARGET_1;
+import static com.commercetools.sync.integration.commons.utils.CartDiscountITUtils.CART_DISCOUNT_TARGET_2;
+import static com.commercetools.sync.integration.commons.utils.CartDiscountITUtils.CART_DISCOUNT_VALUE_1;
+import static com.commercetools.sync.integration.commons.utils.CartDiscountITUtils.CART_DISCOUNT_VALUE_2;
+import static com.commercetools.sync.integration.commons.utils.CartDiscountITUtils.JANUARY_FROM;
+import static com.commercetools.sync.integration.commons.utils.CartDiscountITUtils.JANUARY_UNTIL;
+import static com.commercetools.sync.integration.commons.utils.CartDiscountITUtils.PREDICATE_2;
+import static com.commercetools.sync.integration.commons.utils.CartDiscountITUtils.SORT_ORDER_1;
+import static com.commercetools.sync.integration.commons.utils.CartDiscountITUtils.deleteCartDiscountsFromTargetAndSource;
+import static com.commercetools.sync.integration.commons.utils.CartDiscountITUtils.getCartDiscountByKey;
+import static com.commercetools.sync.integration.commons.utils.CartDiscountITUtils.getSortOrders;
+import static com.commercetools.sync.integration.commons.utils.CartDiscountITUtils.populateSourceProject;
+import static com.commercetools.sync.integration.commons.utils.CartDiscountITUtils.populateTargetProject;
+import static com.commercetools.sync.integration.commons.utils.ITUtils.deleteTypesFromTargetAndSource;
+import static com.commercetools.sync.integration.commons.utils.SphereClientUtils.CTP_TARGET_CLIENT;
+import static io.sphere.sdk.utils.CompletableFutureUtils.exceptionallyCompletedFuture;
+import static java.lang.String.format;
+import static java.util.Collections.singletonList;
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
+
+class CartDiscountSyncIT {
+
+    /**
+     * Deletes types from the target CTP projects.
+     * Populates the target CTP project with test data.
+     */
+    @BeforeEach
+    void setup() {
+        deleteTypesFromTargetAndSource();
```

Comment:
also here doesn't seem that you are using custom fields at all for the integration tests.

Reply:
5e577d9

Reply:
but why rather than removing it, why not use custom fields to have better test coverage of cart discount syncing?